vesc_hw_interface/src/vesc_wheel_controller.cpp

```diff
@@ -53,132 +68,74 @@ void VescWheelController::init(ros::NodeHandle nh, VescInterface* interface_ptr)
   control_timer_ = nh.createTimer(ros::Duration(1.0 / control_rate_), &VescWheelController::controlTimerCallback, this);
 }
 
-void VescWheelController::control(const double target_velocity, const double current_steps, bool reset)
+void VescWheelController::control()
 {
-  if (reset)
+  if (pid_initialize_)
   {
-    target_steps_ = current_steps;
+    if (!sensor_initialize_)
+    {
+      // Start PID control only when sensor initialization is complete
+      pid_initialize_ = false;
+      vesc_step_difference_.getStepDifference(position_steps_, true);
```

Comment:
Since the `reset` flag is called once in both VescWheelController and VescServoController, why don't we add a method in the VescStepDiffference, that resets/initialize `step_in_previous_` and `step_input_queue_`? Or if we know what values to populate beforehand, we can also reset/initialize on the constructor.

This is just a proposal though; I just don't like the reset flag in the `getStepDifference` function for some reason.

Reply:
Add resetStepDifference() and remove the reset argument.